griptape/drivers/vector/qdrant_vector_store_driver.py

```diff
@@ -0,0 +1,212 @@
+
+    def __attrs_post_init__(self) -> None:
+        self.client = import_optional_dependency("qdrant_client").QdrantClient(
+            location=self.location,
+            url=self.url,
+            host=self.host,
+            path=self.path,
+            port=self.port,
+            prefer_grpc=self.prefer_grpc,
+            grpc_port=self.grpc_port,
+            api_key=self.api_key,
+            https=self.https,
+            prefix=self.prefix,
+            force_disable_check_same_thread=self.force_disable_check_same_thread,
+            timeout=self.timeout,
+        )
+
+    def delete_vector(self, vector_id: str) -> None:
+        """
+        Delete a vector from the Qdrant collection based on its ID.
+
+        Parameters:
+            vector_id (str): ID of the vector to delete.
+        """
+        # rest = import_optional_dependency("qdrant_client.http.models")
+        deletion_response = self.client.delete(
+            collection_name=self.collection_name,
+            points_selector=import_optional_dependency("qdrant_client.http.models").PointIdsList(points=[vector_id]),
+        )
+        if deletion_response.status == import_optional_dependency("qdrant_client.http.models").UpdateStatus.COMPLETED:
+            print(f"ID {vector_id} is successfully deleted")
+        else:
+            print(f"Failed to delete ID {vector_id}. Status: {deletion_response.status}")
+
+    def query(
+        self,
+        query: str,
+        count: Optional[int] = None,
+        namespace: Optional[str] = None,
+        include_vectors: bool = False,
+        **kwargs,
+    ) -> list[BaseVectorStoreDriver.Entry]:
+        """
+        Query the Qdrant collection based on a query vector.
+
+        Parameters:
+            query (str): Query string.
+            count (Optional[int]): Optional number of results to return.
+            namespace (Optional[str]): Optional namespace of the vectors.
+            include_vectors (bool): Whether to include vectors in the results.
+
+        Returns:
+            list[BaseVectorStoreDriver.Entry]: List of Entry objects.
+        """
+        query_vector = self.embedding_driver.embed_string(query)
+
+        # Create a search request
+        results = self.client.search(collection_name=self.collection_name, query_vector=query_vector, limit=count)
+
+        # Convert results to QueryResult objects
+        query_results = [
+            BaseVectorStoreDriver.Entry(
+                id=result.id,
+                vector=result.vector if include_vectors else [],
+                score=result.score,
+                meta={k: v for k, v in result.payload.items() if k not in ["_score", "_tensor_facets"]},
+            )
+            for result in results
+        ]
+        return query_results
+
+    def upsert_vector(
+        self,
+        vector: list[float],
+        vector_id: Optional[str] = None,
```

Comment:
```suggestion
        vector_id: Optional[str | int] = None,
```
Can be +ve integers.
https://qdrant.tech/documentation/concepts/points/#point-ids